langserver/fs.py

```diff
@@ -67,7 +67,17 @@ class RemoteFileSystem(FileSystem):
     def __init__(self, conn: JSONRPC2Connection):
         self.conn = conn
 
-    def open(self, path, parent_span):
+    def open(self, path, parent_span=None):
+        if parent_span is None:
+            resp = self.conn.send_request("textDocument/xcontent", {
+                "textDocument": {
+                    "uri": "file://" + path
```

Comment:
Shouldn't you %-encode the path? Looks like this fails if the path contains special characters